tables doc (#3239)
Codecademy/docs: content/luau/concepts/tables/tables.md

---
Title: 'Tables'

> Description: 'luau is a scripting language particularly for Roblox game development. This document is meant to demonstrate the use case for luau tables.'

Reviewer: ```suggestion
Description: 'In the context of luau and Roblox, tables are data types that can store any types of value.'
```

Codecademy Hub Page: 'https://www.codecademy.com/catalog/language/luau'

> CatalogContent:
>   - 'paths/computer-science'
> Subjects:
>   - 'Developer Tools'
>   - 'Game Development'
> Tags:
>   - 'Markdown'
>   - 'GitHub'
>   - 'Documentation'

Reviewer 2: ```suggestion
Subjects:
  - 'Developer Tools'
  - 'Game Development'
Tags:
  - 'Collections'
  - 'Data Structures'
CatalogContent:
  - 'paths/computer-science'
```

---

> The table data type can store multiple values of any type that isn't nil, including booleans, numbers, strings, functions, and other tables.

Reviewer 2: ```suggestion
In the context of Luau and Roblox, **tables** are a data structure that can store multiple values of any type that isn't nil, including booleans, numbers, strings, functions, and other tables.
```

> 
> The syntax for an empty table is:

Reviewer 2: ```suggestion
## Syntax

The syntax for an empty table is:
```

> ```

Reviewer 2: ```suggestion
```lua
```

Reviewer 2: update the other blocks so that they show syntax highlighting

local table = {}
```

> In the context of Lua, the local keyword is primarily used to declare local variables. It is not used specifically for instantiating tables. You can use the local keyword to create local variables of various types, including tables, numbers, strings, functions, and more. What creates the table is the curly braces in the syntax ({}). If you wanted to instantiate a table with parameters you can write something like this:

Reviewer: [proofreading] Since tables are being discussed in a Luau context, consider using that instead of Lua. Also, hyperlink the word to help readers see the connection more. 
```suggestion
In the context of [Luau](https://www.codecademy.com/resources/docs/luau), the local keyword is primarily used to declare local variables. It is not used specifically for instantiating tables. You can use the local keyword to create local variables of various types, including tables, numbers, strings, functions, and more. What creates the table is the curly braces in the syntax ({}). If you want to instantiate a table with parameters you can write something like this:
```

> In the context of [Luau](https://www.codecademy.com/resources/docs/luau), the local keyword is primarily used to declare local variables. It is not used specifically for instantiating tables. You can use the local keyword to create local variables of various types, including tables, numbers, strings, functions, and more. What creates the table is the curly braces in the syntax ({}). If you want to instantiate a table with parameters you can write something like this:

Reviewer 2: ```suggestion
Tables can be defined using curly braces (`{}`). The `local` keyword is primarily used to declare local variables. It is not used specifically for instantiating tables. In order to instantiate a table with elements the syntax will look something like this:
```

Reviewer 2: In docs we avoid the use of first and second person prounouns (I, you, we)

```
local table = {"amy", "john", "bob"} 
```

The above example is instantiated like an [array](https://www.codecademy.com/resources/docs/javascript/arrays), but as mentioned above, table data types are capable of storing any form of data other than nil. Here is another example:

```
local profile = {"john",  24, "Atlanta",  false}
```

Not only can tables be used as arrays, but you can even use them as [dictionaries](https://www.codecademy.com/resources/docs/python/dictionaries). Here is an example of a table in the form of a dictionary with an array data type included:

```
local profile = {
    username = "john_doe",
    level = 12,
    experience = 500,
    items = {"sword", "shield", "knife", "helmet"},
}
```

> 

Reviewer 2: remove empty line. There should never be more than one empty line between content

### Accessing Items

> 

Reviewer 2: ```suggestion
remove empty line
```

> If your table is set up like an array, an important thing to note is that the index begins at 1 rather than at 0 as it is for other languages. Let's use this table as an example:

Reviewer: ```suggestion
If your table is set up like an [array](https://www.codecademy.com/resources/docs/javascript/arrays), an important thing to note is that the index begins at 1 rather than at 0 as it is for other languages. Let's use this table as an example:
```

Author: Alright. I'm not seeing a path for dictionary documentation. Does that not exist yet? lol Should I just create a hyperlink for each datatype?

Reviewer: I linked it in there. Here it is: https://www.codecademy.com/resources/docs/javascript/arrays

Author: Yeah, I included that one when you shared it, but I was going to include a link for my dictionary example because I thought you might mention it.

Author: nvm overlooked it in the python directory

> If your table is set up like an array, an important thing to note is that the index begins at 1 rather than at 0 as it is for other languages. Let's use this table as an example:

Reviewer 2: ```suggestion
If a given table is set up like an array, an important thing to note is that the index begins at `1` rather than at `0` as it is for other languages. Let's use this table as an example:
```

```
 items = {"sword", "shield", "knife", "helmet"}
```

> If you want to print the first item of the table in the console, you'd have to call it like this:

Reviewer 2: ```suggestion
To print the first item of the table to the console, the call will look like this:
```

```
print(items[1])
```

> Calling items from a table when it's formatted like a dictionary is a little different. You can get the value of items by the keys. Below is an example of how you can get the username value from the profile table.

Reviewer 2: ```suggestion
Calling items from a table when it's formatted like a dictionary is a little different. The value of items can be obtained by the keys. Below is an example of how to obtain the `username` value from the `profile` table.
```

```

> //creating the profile table

Reviewer 2: code blocks should use 2 spaces for indentations, not tabs. Comments should have a space and begin with a capital -
// Printing the username in the table

Reviewer 3: Yes I do agree
